Workflow/Sources/RuntimeConfiguration.swift

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/// System for managing configuration options for Workflow runtime behaviors.
+/// - important: These interfaces are subject to breaking changes without corresponding semantic
+/// versioning changes.
+@_spi(WorkflowRuntimeConfig)
+public enum Runtime {
+    @TaskLocal
+    static var _currentConfiguration: Configuration?
+
+    static var _bootstrapConfiguration = BootstrappableConfiguration()
+
+    /// Bootstrap the workflow runtime with the given configuration.
+    /// This can only be called once per process and must be called from the main thread.
+    ///
+    /// - Parameter configuration: The runtime configuration to use.
+    @MainActor
+    public static func bootstrap(
```

Comment:
not at all a blocker, but IMO the one-time bootstrapping requirement is overkill vs having a static var that can be changed at any time